packages/nns/src/token.ts

```diff
@@ -124,11 +124,19 @@ export class TokenAmount {
     amount: number;
     token: Token;
   }): TokenAmount {
-    const e8s = BigInt(Math.floor(amount * Number(E8S_PER_TOKEN)));
-    return TokenAmount.fromE8s({
-      amount: e8s,
+    const tokenAmount = TokenAmount.fromString({
+      amount: amount.toString(),
       token,
     });
+    if (tokenAmount instanceof TokenAmount) {
+      return tokenAmount;
+    }
+    if (tokenAmount === FromStringToTokenError.FractionalMoreThan8Decimals) {
+      throw new Error(`Number ${amount} has more than 8 decimals`);
```

Comment:
I believe it's better api to throw an error here than to return the enum.

This way the usage stays the same and only when we catch errors there might be this one.